### ISSUE TITLE
JavaScript exceptions inside router loop don't get properly caught

### ISSUE THREAD

**Issue by shaharz** [ptaoussanis/sente#97]
![image](https://cloud.githubusercontent.com/assets/205390/5859186/63b68e58-a226-11e4-8458-e3b69d346d56.png)

Being a cljs/core.async newbie it took a couple of hours of head banging to crack this. It turns out my event handler was throwing a JS exception, which currently breaks core.async. This seems to have already been addressed here in an as-of-yet unreleased commit (https://github.com/clojure/core.async/commit/bd71ba73b43ea09e242d37a54eaaa0fed761504c), so I'm not sure there's even a need for a workaround.

Reproduce with, for example:
```clojure
(sente/start-chsk-router! ch-recv #(i-dont-exist))
```
or simply:
```clojure
(go (try (i-dont-exist) (catch :default e e)))
```

**TL;DR: not a Sente issue, still annoying and might trip noobs.**

**Comment by ptaoussanis** [comment]
Hi shaharz, thanks for bringing this up! 

I'd seen the error before, but had never actually traced it to the cause - so that's very welcome.

Would happily merge a Sente PR that adds a workaround clearly marked _as_ a temporary workaround for https://github.com/clojure/core.async/commit/bd71ba73b43ea09e242d37a54eaaa0fed761504c.

Might be a while before core.async is updated (and everyone is on a fixed release).

**Comment by ptaoussanis** [comment]
Will be pushing a workaround to the dev branch shortly.
